### PR TITLE
fix types action (test)

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: .github/workflows/types.yaml [hunk 1]
```diff
@@ -10,111 +10,162 @@ jobs:
...
+            MINOR=$((MINOR + 1))
+            PATCH=0
+          elif [[ "$VERSION_FRAGMENT" == "patch" ]]; then
+            PATCH=$((PATCH + 1))
+          elif [[ "$VERSION_FRAGMENT" == "beta" ]]; then
+            PATCH=$((PATCH + 1))
+            NEW_VERSION="${MAJOR}.${MINOR}.${PATCH}-beta"
+          else
+            NEW_VERSION="${MAJOR}.${MINOR}.${PATCH}"
+          fi
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/types.yaml:143]: good suggestion. I've added it in a slightly different way

File: .github/workflows/types.yaml [hunk 2]
```diff
@@ -10,111 +10,162 @@ jobs:
...
-                if [[ "$tag_name" == "$LATEST_TAG-$beta"* ]]; then
-                  LATEST_TAG=$tag_name
+              while read ITEM; do
+                TAG_NAME=$(jq -r '.name' <<< "$ITEM")
+                if [[ "$TAG_NAME" == "$LATEST_TAG-$beta"* ]]; then
+                  LATEST_TAG=$TAG_NAME
                   break
                 fi
-              done <<<$(echo "$release_json" | jq -c -r '.[]')
+              done <<<$(echo "$RELEASE_JSON" | jq -c -r '.[]')
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/types.yaml:87]: added

File: .github/workflows/types.yaml [hunk 3]
```diff
@@ -10,111 +10,162 @@ jobs:
...
           ssh-key: ${{ secrets.SSH_REPO_TOKEN }}
           ref: main
 
-      - name: Copy types bindings
+      # Step 9: Copy generated types and ABI into the lifi-contract-types repo
+      - name: Copy Type Bindings and ABI
         run: |
           rm -r lifi-contract-types/src/
           mv typechain lifi-contract-types/src
           cp diamond.json lifi-contract-types/dist/
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/types.yaml:62]: works like this, it's fine. Not needed

File: .github/workflows/types.yaml [hunk 4]
```diff
@@ -10,111 +10,162 @@ jobs:
...
-
-          # Sanitize the commit message by removing single quotes
-          COMMIT_MSG=$(echo "$MESSAGE" | sed "s/'//g")
+          git commit -m "actions: new contracts version $NEW_VERSION"
 
-          git tag -a v${{ steps.bump_version.outputs.next-version }} -m "$MESSAGE"
-          git push origin tag v${{ steps.bump_version.outputs.next-version }}
+          # Annotate and push the new tag
+          git tag -a "v$NEW_VERSION" -m "$MESSAGE"
+          git push origin tag "v$NEW_VERSION"
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/types.yaml:168]: works as-is...

**Review by ezynda3 — APPROVED** [review]

File: .github/workflows/types.yaml [hunk 5]
```diff
@@ -10,111 +10,171 @@ jobs:
...
-              while read item; do
-                tag_name=$(jq -r '.name' <<< "$item")
-                # check if there's already a latest tag beta release
-                # and, if present, use it instead of the main one
-                # if we end up without any latest beta tag, we will create a beta release from the latest tag
-                if [[ "$tag_name" == "$LATEST_TAG-$beta"* ]]; then
-                  LATEST_TAG=$tag_name
+              while read ITEM; do
+                TAG_NAME=$(jq -r '.name' <<< "$ITEM")
+                if [[ "$TAG_NAME" == "$LATEST_TAG-$beta"* ]]; then
```
Comment (Matt-LiFi) [inline .github/workflows/types.yaml:83]: Dont know if you mean to but $beta is in lowercase here.

**Review by Matt-LiFi — APPROVED** [review]

File: .github/workflows/types.yaml (hunk 5 above)
Comment (0xDEnYO) [reply .github/workflows/types.yaml:83]: works like this...seems fine

**Pull request merged**